test/server/config_validation/server_test.cc

```diff
@@ -11,23 +13,60 @@ namespace Server {
 // Test param is the path to the config file to validate.
 class ValidationServerTest : public testing::TestWithParam<std::string> {
 public:
-  static void SetUpTestCase() {
+  static void SetupTestDirectory() {
     TestEnvironment::exec(
         {TestEnvironment::runfilesPath("test/config_test/example_configs_test_setup.sh")});
     directory_ = TestEnvironment::temporaryDirectory() + "/test/config_test/";
   }
 
+  static void SetUpTestCase() { SetupTestDirectory(); }
+
 protected:
   ValidationServerTest() : options_(directory_ + GetParam()) {}
 
   static std::string directory_;
-
   testing::NiceMock<MockOptions> options_;
   TestComponentFactory component_factory_;
 };
 
 std::string ValidationServerTest::directory_ = "";
 
+// ValidationServerTest_1 is created only to run different set of parameterized
+// tests than set of tests for ValidationServerTest.
+class ValidationServerTest_1 : public ValidationServerTest {
+public:
+  static const std::vector<std::string> GetAllConfigFiles() {
+    SetupTestDirectory();
+
+    std::string all_file_names =
+        TestEnvironment::readFileToStringForTest(directory_ + "test_out/all_config_files.txt");
+    std::vector<std::string> files;
+    // all_config_files.txt contains a list of all config files from tar file created by
+    // example_configs_test_setup.sh Names of the files are separated by EOL. Tokenize those  names
+    // and put them into a std container.
+    size_t pos = 0;
+    size_t prev_pos = pos;
+    while (pos != std::string::npos) {
+      pos = all_file_names.find('\n', pos);
+      if (pos != std::string::npos) {
+        files.emplace_back(all_file_names.substr(prev_pos, pos - prev_pos));
+        pos += 1;
+        prev_pos = pos;
+      } else {
+        // last item. Process it if not empty line.
+        if (prev_pos < all_file_names.length()) {
+          files.emplace_back(all_file_names.substr(prev_pos));
+        }
+      }
+    }
```

Comment:
This is secondary to @mattklein123's questions, but I believe you can replace this loop with
```
std:vector<std::string> files = absl::StrSplit(all_file_names, '\n', absl::SkipEmpty());
```

(There's also Utility::splitToken, but it returns a vector of absl::string_view, which you'd have to convert to std:strings since all_file_names doesn't exist beyond this function's scope.)

Reply:
@zuercher Thanks for reviewing. If I cannot use the common code, I will replace my tokenizing procedure with code you suggest.